### PR TITLE
fix the bug: Cannot run from src/test anymore bug #413

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

File: com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/ResolveClasspathsHandler.java [hunk 1]
```diff
@@ -200,10 +208,73 @@ public void acceptSearchMatch(SearchMatch match) {
...
+            for (int i = 0; i < packageFragmentRoot.length; i++) {
+                if (packageFragmentRoot[i].getElementType() == IJavaElement.PACKAGE_FRAGMENT_ROOT
+                        && packageFragmentRoot[i].getKind() == IPackageFragmentRoot.K_SOURCE) {
+                    IClasspathEntry cpe = packageFragmentRoot[i].getResolvedClasspathEntry();
+                    if (forTestOnly(cpe)) {
+                        testFolders.add(packageFragmentRoot[i]);
+                    }
+                }
+            }
+        } catch (JavaModelException e) {
```
Comment (testforstephen) [inline com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/ResolveClasspathsHandler.java]: extract the logic to a function `getTestFolders(IJavaProject project)`.

File: com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/ResolveClasspathsHandler.java [hunk 2]
```diff
@@ -200,10 +208,73 @@ public void acceptSearchMatch(SearchMatch match) {
...
+            // ignore
+        }
+        if (!testFolders.isEmpty()) {
+            try {
+
+                List<Object> mainClassesInTestFolder = new ArrayList<>();
```
Comment (testforstephen) [inline com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/ResolveClasspathsHandler.java]: boolean isFound = false;

File: com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/ResolveClasspathsHandler.java [hunk 3]
```diff
@@ -200,10 +208,73 @@ public void acceptSearchMatch(SearchMatch match) {
...
+                        SearchPattern.R_CASE_SENSITIVE | SearchPattern.R_EXACT_MATCH);
+                SearchEngine searchEngine = new SearchEngine();
+                IJavaSearchScope scope = SearchEngine
+                        .createJavaSearchScope(testFolders.toArray(new IJavaElement[0]), constraints);
+                SearchRequestor requestor = new SearchRequestor() {
+                    @Override
+                    public void acceptSearchMatch(SearchMatch match) {
+                        Object element = match.getElement();
+                        if (element instanceof IJavaElement) {
+                            mainClassesInTestFolder.add(element);
```
Comment (testforstephen) [inline com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/ResolveClasspathsHandler.java]: isFound = true;

File: com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/ResolveClasspathsHandler.java (hunk 2 above)
Comment (andxu) [reply com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/ResolveClasspathsHandler.java]: I cannot change the value isFound at the callback

**Pull request merged**